tests/components/wake_on_lan/test_switch.py

```diff
@@ -1,266 +1,287 @@
+            blocking=True,
         )
-        self.hass.block_till_done()
 
-        state = self.hass.states.get("switch.wake_on_lan")
-        assert STATE_OFF == state.state
-
-        TEST_STATE = True
+        state = hass.states.get("switch.wake_on_lan")
+        assert STATE_ON == state.state
 
-        common.turn_on(self.hass, "switch.wake_on_lan")
-        self.hass.block_till_done()
+        await hass.services.async_call(
+            switch.DOMAIN,
+            SERVICE_TURN_OFF,
+            {ATTR_ENTITY_ID: "switch.wake_on_lan"},
+            blocking=True,
+        )
 
-        state = self.hass.states.get("switch.wake_on_lan")
+        state = hass.states.get("switch.wake_on_lan")
         assert STATE_ON == state.state
 
-        common.turn_off(self.hass, "switch.wake_on_lan")
-        self.hass.block_till_done()
 
-        state = self.hass.states.get("switch.wake_on_lan")
-        assert STATE_ON == state.state
+async def test_valid_hostname_windows(hass):
+    """Test with valid hostname on windows."""
+    assert await async_setup_component(
+        hass,
+        switch.DOMAIN,
+        {
+            "switch": {
+                "platform": "wake_on_lan",
+                "mac": "00-01-02-03-04-05",
+                "host": "validhostname",
+            }
+        },
+    )
+    await hass.async_block_till_done()
+
+    state = hass.states.get("switch.wake_on_lan")
+    assert STATE_OFF == state.state
+
+    with patch.object(subprocess, "call", return_value=0), patch.object(
+        platform, "system", return_value="Windows"
+    ):
+        await hass.services.async_call(
+            switch.DOMAIN,
+            SERVICE_TURN_ON,
+            {ATTR_ENTITY_ID: "switch.wake_on_lan"},
+            blocking=True,
+        )
 
-    @patch("wakeonlan.send_magic_packet", new=send_magic_packet)
-    @patch("subprocess.call", new=call)
-    @patch("platform.system", new=system)
-    def test_valid_hostname_windows(self):
-        """Test with valid hostname on windows."""
-        global TEST_STATE
-        TEST_STATE = False
-        assert setup_component(
-            self.hass,
+    state = hass.states.get("switch.wake_on_lan")
+    assert STATE_ON == state.state
+
+
+async def test_broadcast_config_ip_and_port(hass):
+    """Test with broadcast address and broadcast port config."""
+    mac = "00-01-02-03-04-05"
+    broadcast_address = "255.255.255.255"
+    port = 999
+
+    assert await async_setup_component(
+        hass,
+        switch.DOMAIN,
+        {
+            "switch": {
+                "platform": "wake_on_lan",
+                "mac": mac,
+                "broadcast_address": broadcast_address,
+                "broadcast_port": port,
+            }
+        },
+    )
+    await hass.async_block_till_done()
+
+    state = hass.states.get("switch.wake_on_lan")
+    assert STATE_OFF == state.state
+
+    with patch.object(
+        wakeonlan, "send_magic_packet", new=send_magic_packet
```

Comment:
You can remove the patching of `send_magic_packet` everywhere. It's now automatically applied.